### ISSUE TITLE
Chat Data

### ISSUE THREAD

**Issue by Martijnpold** [team-avans-p/catan#6]
The Chat Data classes are all the classes needed to handle chat and log messages. These data classes save all the messages as seperate "Message" instances which have a unique ID, a timestamp and content.

This includes the classes:
- ChatData
- LogData
- Message
- LogMessage
- ChatMessage

**Comment by Stijnwingens** [comment]
Proposition:
1. Create all of the classes
2. Create all of the instance variables
3. Create all of the constructors
4. Don't know yet...

**Comment by Stijnwingens** [comment]
We only need methods to save messages to, and get messages from the database in these classes right? The rest is all done in the constructors, right? @Martijnpold ... I guess the methods for saving and getting messages can also be created after we have established a connection with the database?

**Comment by jessevdp** [comment]
OK so first, we need the variables, constructor, and getters. Connection with the database will be established later. We have a separate issue for that (#4). Once that's done we can couple the data classes and the database via SQL queries.

**Comment by Stijnwingens** [comment]
@Martijnpold @jessevdp can someone explain to me what will be done in the ChatData and LogData classes? And I am presuming that LogMessage and ChatMessage extend from Message?

**Comment by Martijnpold** [comment]
Basically the ChatData and LogData have to be able to convert the data from the database into instances of the object LogMessage and ChatMessage. These can then be requested from the controller.

**Comment by Stijnwingens** [comment]
and the LogMessage and ChatMessage extend from Message?

**Comment by Stijnwingens** [comment]
should the getMessage from the Message class refer to the ChatData/LogData class to get the message from the database then? Because I'm not sure anymore whether Message should have a getter for the message content if it just calls another class to get the message... @Martijnpold?

**Comment by Martijnpold** [comment]
It'd be best to have the instance variables set to private and to give them getters for all the instance variables. They don't really need anything else since there is no reason to "change" a message once it's been sent.

**Comment by jessevdp** [comment]
@Stijnwingens 

> and the LogMessage and ChatMessage extend from Message?

If you check the `Application Class Diagram` you can see that both `ChatMessage` and `LogMessage` are extensions of the abstract class `Message`.

The other part of your question doesn't make much sense to be honest.

**Comment by Stijnwingens** [comment]
I have that @Martijnpold but the getters will only request the message from another getter in ChatData/LogData... Why make the getter in the Message class at all then? Why not just create a getter in ChatData/ LogData and as parameter the id of the message you want to get? @jessevdp @Martijnpold

**Comment by Stijnwingens** [comment]
@Martijnpold what type should the getMessages() return?

**Comment by Stijnwingens** [comment]
I just thought of the fact that ChatData and LogData need to be able to add messages to the db ... Right?

**Comment by Stijnwingens** [comment]
Or is this done by the controller?

**Comment by jessevdp** [comment]
@Stijnwingens Yes, these classes should be able to connect to the database. However since we don't have access to the database yet we can't add this functionality.

Perhaps we should make some new issues on the matter of connecting data classes to the database.
